taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -328,24 +374,37 @@ const ListTranslations = ({ nodeObject, setNodeObject }) => {
                         if (!language) language = '';
                         setNewLanguageCode(ISO6391.getCode(language));
                         const isValidLanguage = ISO6391.validate(ISO6391.getCode(language))
+                        const isAlreadyShown = shownTranslationLanguages.includes(ISO6391.getCode(language))
                         const isDuplicateLanguage = renderedTranslations.some(el => (el.languageCode === ISO6391.getCode(language))) || 
                                                 nodeObject.main_language === ISO6391.getCode(language)
-                        if (isValidLanguage && !isDuplicateLanguage) {setisValidLanguageCode(true)}
-                        else {setisValidLanguageCode(false)}
+                        if (isValidLanguage  && !isDuplicateLanguage) {setLanguageAction("add")}
+                        else if (isValidLanguage && !isAlreadyShown && isDuplicateLanguage) {setLanguageAction("show")}
+                        else {setLanguageAction("invalid")}
                     }}
-                    renderInput={(params) => 
+                    renderOption={(props, option) => {
+                        const languageCode = ISO6391.getCode(option)
+                        const isAlreadyShown = addedLanguageCodes.includes(languageCode)
+                        // set color of text to green if the language is already shown
+                        return (
+                            <span {...props} style={{ color: isAlreadyShown ? 'green' : 'black' }}>
+                                {isAlreadyShown ? <b>{option}</b> : option}
+                            </span>
```

Comment:
If I'm correct, it's not if it's already shown but if it already have a translation.

So you may change comment and variable name.

Reply:
True, will do.

Reply:
Actually, it will be green for one more case,
when the user added a language, but did not add a translation yet, so I am renaming it to `presentInObjectNode`